### PR TITLE
Draft PR for VyOS init script is out of date #885

### PR DESCRIPTION
**Pull request by Giggum**
Draft PR for VyOS init script is out of date #885 (https://github.com/nextdns/nextdns/issues/885)

### CODE REVIEW

File: host/service/vyos/service.go [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+	Path string
+	PostCfg_Script string
+}
+
+func New(c service.Config) (Service, error) {
+	if st, err := os.Stat("/config/scripts/"); err != nil || !st.IsDir() {
+		if _, err = os.Stat("/usr/libexec/vyos/init/vyos-router"); err != nil {
+			return Service{}, service.ErrNotSupported
+		}
+	}
```
Comment (rs) [inline host/service/vyos/service.go:29]:
```suggestion
func isVyOS() bool {
	if st, err := os.Stat("/config/scripts/"); err != nil || !st.IsDir() {
		if _, err = os.Stat("/usr/libexec/vyos/init/vyos-router"); err != nil {
			return true
		}
	}
	return false
}

func New(c service.Config) (Service, error) {
	if !isVyOS() {
		return Service{}, service.ErrNotSupported
	}
```

Comment (rs) [reply host/service/vyos/service.go:29]: This would probably fix your issue.

**Comment by rs** [comment]
Did you test this version and is confirmed working? If so, please update the import path and remove the draft flag so I can merge.

**Comment by Giggum** [comment]
Hi @rs I opened a new Draft PR as I believe this is a better approach to take. I've tested the install.sh script by pointing to my fork: 

'/bin/sh -c '/bin/sh -c "$(curl -sL https://raw.githubusercontent.com/Giggum/nextdns/add-vyos-885_2/install.sh)"'

Install script is good, but the changes made to my forked import path are not appearing as it's defaulting to use github.com/nextdns/nextdns/ rather than github.com/Giggum/nextdns/add-vyos-885_2/

**Comment by rs** [comment]
If you don’t push the go.mod change and keep all the module to nextdns/nextdns, our CI will build a snapshot you will be able to test e2e.

**Comment by Giggum** [comment]
Closing this PR in lieu of #921

**Comment by Giggum** [comment]
> If you don’t push the go.mod change and keep all the module to nextdns/nextdns, our CI will build a snapshot you will be able to test e2e.

Okay I understand the workflow now. I was under the impression that I could test locally before opening a PR and was failing at it miserably. I thought if I cloned my fork onto my vyos router, I could run the modified install.sh along with any changes I introduced and test that way.

**Pull request closed without merging**